nemo/collections/llm/recipes/llama3_70b.py

```diff
@@ -0,0 +1,113 @@
+from typing import Callable, Optional
+
+import pytorch_lightning as pl
+import torch
+from megatron.core.distributed import DistributedDataParallelConfig
+from pytorch_lightning.callbacks.callback import Callback
+
+from nemo import lightning as nl
+from nemo.collections.llm.api import finetune, pretrain
+from nemo.collections.llm.gpt.data.mock import MockDataModule
+from nemo.collections.llm.gpt.data.squad import SquadDataModule
+from nemo.collections.llm.gpt.model.llama import Llama3Config70B, LlamaModel
+from nemo.collections.llm.peft.lora import LoRA
+from nemo.collections.llm.recipes.log.default import default_log, default_resume, tensorboard_logger
+from nemo.collections.llm.recipes.optim.adam import distributed_fused_adam_with_cosine_annealing
+from nemo.collections.llm.utils import Config, Partial
+from nemo.utils.exp_manager import TimingCallback
+
+NAME = "llama3_70b"
+
+
+def model() -> Config[pl.LightningModule]:
+    return Config(LlamaModel, config=Config(Llama3Config70B))
+
+
+def trainer(
+    tensor_parallelism: int,
+    pipeline_parallelism: int,
+    pipeline_parallelism_type: Optional[torch.dtype],
+    virtual_pipeline_parallelism: Optional[int],
+    context_parallelism: int,
+    sequence_parallelism: bool,
+    num_nodes: int = 1,
+    num_gpus_per_node: int = 8,
+    max_steps: int = 1168251,
+    callbacks: Optional[list[Config[Callback]]] = None,
+) -> Config[nl.Trainer]:
+    strategy = Config(
+        nl.MegatronStrategy,
+        tensor_model_parallel_size=tensor_parallelism,
+        pipeline_model_parallel_size=pipeline_parallelism,
+        pipeline_dtype=pipeline_parallelism_type,
+        virtual_pipeline_model_parallel_size=virtual_pipeline_parallelism,
+        context_parallel_size=context_parallelism,
+        sequence_parallel=sequence_parallelism,
+        gradient_as_bucket_view=True,
+        ckpt_include_optimizer=True,
+        ckpt_async_save=True,
+        ckpt_parallel_load=True,
+        ddp=Config(
+            DistributedDataParallelConfig,
+            check_for_nan_in_grad=True,
+            grad_reduce_in_fp32=True,
+        ),
+    )
+
+    trainer = Config(
+        nl.Trainer,
+        accelerator="gpu",
+        accumulate_grad_batches=1,
+        callbacks=callbacks,
+        devices=num_gpus_per_node,
+        gradient_clip_val=1.0,
+        limit_test_batches=50,
+        limit_val_batches=32,
+        log_every_n_steps=10,
```

Comment:
Default to 1 so steps are not skipped?

Reply:
Yeah I can change it, but could become a bit verbose. Also not sure if it will have any impact on performance.

Reply:
Since nemo 1.0 also uses 10, makes sense to skip steps and reduce verbosity. Thanks